README.md

```diff
@@ -71,6 +89,106 @@ To get around this, you can create an
+```
+
+In another terminal:
+
+```
+cd examples/simple # or any other application directory
+npm install
+rm -rf node_modules/react node_modules/react-dom node_modules/react-plotly.js-edit
+npm link react-plotly.js-edit
+npm start # keep this running
+```
+
+## Built-in Components
+
+This module provides a number of nestable _containers_ which are intended to contain _fields_ that render _widgets_ that have been connected to individual values in the figure via _connector functions_. Containers can also be connected to certain parts of the figure tree such that their child fields map to the appropriate leaf values. A field must have a connected container as an ancestor in order to be bound to the figure. The `<PlotlyEditor />` and connector functions use the React [`context` API]() to push configuration information to child components.
+
+At a pseudo-code level it looks like this:
+
+```javascript
+<PlotlyEditor {...etc}>
+  <ConnectedContainer {...etc}>
+    <Field attr="path.to.figure.value" {...etc} />
+  </ConnectedContainer>
+</PlotlyEditor>
+```
+
+The [custom editor example](https://github.com/plotly/react-plotly.js-editor/tree/master/examples/custom) shows how to build a custom editor, and shows off all of the general-purpose containers and fields listed below.
+
+### General-purpose Containers
+
+* `<Panel />`: renders as a generic rectangular container with special handling for collapsing/expanding child `<Fold />`s and optionally an 'add' button for creating them
+* `<PanelMenuWrapper />`: renders as a sidebar selector menu for `<Panel />`s
+* `<Fold />`: collapsable container within a `<Panel />`
+* `<Section />`: uncollapsable container within a `<Panel />` or `<Fold />`
+* `<MenuPanel />`: container child of `<Section />` that appears when a cog icon in the section title is clicked on
+
+### General-purpose Fields
+
+Fields all accept an `attr` property (if bindable) to indicate which figure value they are bound to. This property can be a `.`-delimited path to a leaf, starting at the context-appropriate point in the figure for the parent container (see connector functions below).
+
+* `<Numeric />`: renders as a text field with arrows and units, useful for numeric values
+* `<Radio />`: renders as a button group, useful for mutually-exclusive low-cardinality enumerable values
+* `<Dropdown />`: renders as a dropdown menu useful for mutually-exclusive high-cardinality enumerable values
+* `<ColorPicker />`: renders as a popup color-picker, useful for CSS color hex value strings
+* `<Flaglist />`: renders as a list of checkboxes, useful for `+`-joined flag lists like `data[].mode`
+* `<MultiFormatTextEditor />`: renders as a WYSIWYG editor, useful for text like `layout.title`
+* `<Info />`: renders as text, useful for displaying help text (unbindable)
+
+<p align="center">
+  <img src="examples/custom/components.png" alt="Components" width="432" height="692" border="1">
+</p>
+
+### Connector functions
```

Comment:
I'm tempted to move `Connector functions` to the bottom of this section. Connector functions are only needed when building your own containers which can be considered "advanced". 
Or perhaps we have `standard usage` and `advanced usage` sections. 
Just thinking out loud.

Reply:
See above: until we export some of the pre-connected components like `LayoutPanel` and `AxesFold` I think we are stuck with documenting these somewhat like this, otherwise people can't build all that much.